carbon-design-system/carbon
# feat(TimePickerSelect): updated props for v11

REF #9535 #9624 

Refactoring `TimePickerSelect` with prop changes in preparation for migration to a functional component.

#### Storybook

- Removed `iconDescription` from "Select" in the `TimePicker` story.
- Added an `aria-label` knob to "Select" in the `TimePicker` story.

#### TimePickerSelect

- Added a deprecation warning to `iconDescription`.
- Remove the unused `inline` prop.
- Put the props in alphabetical order.
- Renamed `...other` -> `...rest`
- Added an `ariaLabel` prop.
- Added proper aria attributes to the Chevron.

#### Testing
- Make sure `TimePicker` tests are passing.
- Ensure `TimePicker` does it's thing in Storybook, especially as it pertains to `TimePickerSelect`.
